pennylane_lightning/src/Gates.cpp

```diff
@@ -407,6 +446,16 @@ void Pennylane::CRotationXGate::applyKernel(const StateVector& state, const std:
     }
 }
 
+const double Pennylane::CRotationXGate::generatorScalingFactor{ -0.5 };
+
+void Pennylane::CRotationXGate::applyGenerator(const StateVector& state, const std::vector<size_t>& indices, const std::vector<size_t>& externalIndices) {
+    for (const size_t& externalIndex : externalIndices) {
+        CplxType* shiftedState = state.arr + externalIndex;
+        shiftedState[indices[0]] = shiftedState[indices[1]] = 0;
+        swap(shiftedState[indices[2]], shiftedState[indices[3]]);
+    }
```

Comment:
Optionally, you could also directly apply `CNOT` here

Reply:
Is that preferable? I would think a swap would be faster. We still need to set `shiftedState[indices[0]]` and `shiftedState[indices[1]]` to 0. 🤔

Reply:
No you're right! In my head, the generator was exactly the CNOT matrix. But yes it's actually `|1><1| otimes X`! So agree that it's good as-is.